### PR TITLE
plutus-pab: Refactor runChainEffects

### PR DESCRIPTION
**Pull request by gilligan**
Refactor Cardano.Node.Mock.runChainEffects slightly to make it easier to
follow what is going on.

I've struggled parsing `runChainEffects` multiple times and rearranging
it this way has definitely helped me. With the extracted where clauses
you can nicely inspect the types and see how they individually run/peel
off effects. Switching to `(&)` the code can now also be read in the
order of sequential execution.

*TL&DR*: Nothing big, but helps with readability (I hope).

Pre-submit checklist:
- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [ ] Reviewer requested
- If you updated any cabal files or added Haskell packages:
    - [ ] `nix-shell shell.nix --run updateMaterialized` to update the materialized Nix files
    - [ ] Update `hie-*.yaml` files if needed
- If you changed any Haskell files:
    - [ ] `nix-shell shell.nix --run fix-stylish-haskell` to fix any formatting issues
- If you changed any Purescript files:
    - [ ] `nix-shell shell.nix --run fix-purty` to fix any formatting issues

Pre-merge checklist:
- [ ] Someone approved it
- [ ] Commits have useful messages
- [ ] Review clarifications made it into the code
- [ ] History is moderately tidy; or going to squash-merge

### CODE REVIEW

File: plutus-pab/src/Cardano/Node/Mock.hs [hunk 1]
```diff
@@ -115,29 +119,32 @@ runChainEffects ::
...
+            & runStderrLog
+            & runM
     liftIO $ putMVar stateVar newState
     pure (events, a)
+        where
+            processBlock e = e >>= \r -> Chain.processBlock >> pure r
+
+            runRandomTx = subsume . runGenRandomTx
+
+            runNodeFollwer = interpret (mapLog FollowerMsg) . FE.handleNodeFollower
```
Comment (gilligan) [inline plutus-pab/src/Cardano/Node/Mock.hs:139]: I really wanted to add short concise type signatures to these bindings indicating which effects are being peeled off. Something sort of like the following:

```haskell
runRandomTx :: (Member GenRandomTx  effs) => Eff (GenRandomTx ': effs) ~> Eff effs
runRandomTx = _
```

But i guess i would have to use the explicit `NodeServerEffects` type somehow and then..all of this doesn't really work. Anyway, beats me.

Comment (j-mueller) [reply plutus-pab/src/Cardano/Node/Mock.hs:139]: Yeah that's the problem with defining lists of effects :(

**Comment by j-mueller** [comment]
Personally I'd prefer to write `interpret @X`  so that you can see which effect gets handled at that place (also helps with type inference if you change something and add a hole to see what goes where). But this is also an improvement.

File: plutus-pab/src/Cardano/Node/Mock.hs (hunk 1 above)
Comment (j-mueller) [reply plutus-pab/src/Cardano/Node/Mock.hs:139]: With the type application after `interpret` you get minimal noise - just the effect that is handled here.

**Review by j-mueller — APPROVED** [review]

**Comment by gilligan** [comment]
> But this is also an improvement.

I'll take that then. 

**Happy to iterate further though** - It would be nice if we could establish something that everyone is happy with to use it everywhere (yay consistency).

**Pull request merged**